data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -99,8 +100,14 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
             throw new PatternSyntaxException("delete_value_regex is not a valid regex string", keyValueProcessorConfig.getDeleteValueRegex(), -1);
         }
 
-        if (keyValueProcessorConfig.getIncludeKeys() != null) {
-            includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
+        includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
+        excludeKeysSet.addAll(keyValueProcessorConfig.getExcludeKeys());
+
+        Set<String> intersectionSet = new HashSet<String>(includeKeysSet);
+        if (intersectionSet.retainAll(excludeKeysSet)) {
```

Comment:
Seems to me this will return true if this set changed as a result of the call, which is not relevant to our validation. i.e. we could have includeKeysSet unchanged as a result of this call but still have complete overlap with excludeKeySet

Reply:
My idea was to create a copy of the includeKeysSet and compare that with excludeKeysSet, since retainAll() will modify the set. If the set is changed as a result of the call, that means that there is some overlap. I didn't want to completely overwrite includeKeysSet in case the original data was still needed somewhere

Reply:
I think the outer if condition can be removed
```suggestion
        intersectionSet.retainAll(excludeKeysSet)
        if (!intersectionSet.isEmpty()) {
```